source/solvers/gls_navier_stokes.cc

```diff
@@ -1852,12 +1852,11 @@ GLSNavierStokesSolver<dim>::solve_linear_system(const bool initial_step,
 
 template <int dim>
 void
-GLSNavierStokesSolver<dim>::setup_ILU()
+GLSNavierStokesSolver<dim>::setup_ILU(int temp_precond_fill_ilu)
 {
   TimerOutput::Scope t(this->computing_timer, "setup_ILU");
 
-  const double ilu_fill =
-    this->simulation_parameters.linear_solver.ilu_precond_fill;
+  const double ilu_fill =temp_precond_fill_ilu;
```

Comment:
Is this variable still necessary? Why not just use the function argument?

Reply:
I actually don't know I think this is a conversion from int to double  since the parameter fill returns an int.

Reply:
Can you verify if the function requires a double or an int? If it requires an int as an argument, it might be that the double is dynamically casted to an int. If that's the case, it should be an int from the get go and that just means I made a stupid mistake in the past.